protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,339 @@
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { TempleMath } from "contracts/common/TempleMath.sol";
+import { EpochLib } from "contracts/templegold/EpochLib.sol";
+import { IAuctionBase } from "contracts/interfaces/templegold/IAuctionBase.sol";
+
+/** 
+ * @title AuctionEscrow
+ * @notice Bidding token is deposited into this contract to bid on a share of distributed Temple Gold for an epoch.
+ * Temple Gold acquired in past epochs can always be claimed. Once bid, users cannot
+ * withdraw their bid token and can claim their share of Temple Gold for epoch after auction finishes.
+ * Elevated access can change bidding token for future epochs.
+ */
+contract DaiGoldAuction is IDaiGoldAuction, AuctionBase, TempleElevatedAccess {
+    using SafeERC20 for ITempleGold;
+    using SafeERC20 for IERC20;
+    using TempleMath for uint256;
+    using EpochLib for IAuctionBase.EpochInfo;
+    
+    /// @notice Temple GOLD address
+    ITempleGold public immutable override templeGold;
+    /// @notice Token to bid for Temple GOLD
+    /// @dev This token is assumed to not have any taxes or complicated mechanisms. Plan is to use DAI token. This could change in future
+    IERC20 public override bidToken;
+    /// @notice Destination address for proceeds of auctions
+    address public immutable override treasury;
+    /// @notice Address that can trigger start of auction. address(0) means anyone
+    address public override auctionStarter;
+
+    /// @notice Keep track of next epoch auction Temple Gold amount
+    uint256 public override nextAuctionGoldAmount;
+
+    /// @notice Auction duration
+    uint64 public constant AUCTION_DURATION = 1 weeks;
+    /// @notice Configuration for auction
+    AuctionConfig private auctionConfig;
+
+    constructor(
+        address _templeGold,
+        address _bidToken,
+        address _treasury,
+        address _rescuer,
+        address _executor,
+        address _auctionStarter
+    ) TempleElevatedAccess(_rescuer, _executor) {
+        templeGold = ITempleGold(_templeGold);
+        bidToken = IERC20(_bidToken);
+        treasury = _treasury;
+        auctionStarter = _auctionStarter;
+        emit AuctionStarterSet(_auctionStarter);
+    }
+
+    /**
+     * @notice Set auction configuration
+     * @param _config Auction configuration
+     */
+    function setAuctionConfig(AuctionConfig calldata _config) external override onlyElevatedAccess {
+        if (_config.auctionStartCooldown == 0
+                || _config.auctionMinimumDistributedGold == 0
+                || _config.auctionsTimeDiff == 0) 
+            { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        if (!epochs[_currentEpochId].hasEnded()) { revert InvalidOperation(); }
+        auctionConfig = _config;
+
+        emit AuctionConfigSet(_currentEpochId, _config);
```

Comment:
tiny gas saving:

`_currentEpochId` is read from storage two times in this function .